pkg/ipamplugin/ipam_plugin.go

```diff
@@ -311,21 +311,27 @@ func cmdAdd(args *skel.CmdArgs) error {
 		}
 
 		if num4 == 1 {
-			if len(assignedV4) != num4 {
-				return fmt.Errorf("failed to request %d IPv4 addresses. IPAM allocated only %d", num4, len(assignedV4))
+			if err := v4Assignments.PartialFulfillmentError(); err != nil {
+				return fmt.Errorf("failed to request IPv4 addresses: %w", err)
 			}
-			ipV4Network := net.IPNet{IP: assignedV4[0].IP, Mask: assignedV4[0].Mask}
+			if len(v4Assignments.IPs) < num4 {
+				return fmt.Errorf("failed to request %d IPv4 addresses. IPAM allocated only %d", num4, len(v4Assignments.IPs))
+			}
```

Comment:
will we ever get to this case now? I thought the `PartialFullfillmentError` is only populated if this same condition was hit?

Reply:
Right, I don't think we should, but I was trying to be safe. (Is that "too safe"? Should I remove it then?)

Reply:
Will remove it!